runbot/runbot.py

```diff
@@ -1126,7 +1126,10 @@ def _schedule(self, cr, uid, ids, context=None):
         default_timeout = int(icp.get_param(cr, uid, 'runbot.timeout', default=1800)) / 60
 
         for build in self.browse(cr, uid, ids, context=context):
-            if build.state == 'pending':
+            if build.state == 'deathrow':
+                build._kill(result='killed')
```

Comment:
didn't want to create a new result type, but I guess you're right, I'll change it.